src/test/spec/unified_runner/mod.rs

```diff
@@ -43,25 +43,25 @@ pub(crate) fn run_unified_tests(spec: &'static [&'static str]) -> RunUnifiedTest
 type FileTransformation = Box<dyn Fn(&mut TestFile) + Send + Sync>;
 pub(crate) struct RunUnifiedTestsAction {
     spec: &'static [&'static str],
-    skipped_files: Option<&'static [&'static str]>,
-    skipped_tests: Option<&'static [&'static str]>,
+    skipped_files: Option<Vec<&'static str>>,
+    skipped_tests: Option<Vec<&'static str>>,
     file_transformation: Option<FileTransformation>,
 }
 
 impl RunUnifiedTestsAction {
     /// The files to skip deserializing. The provided filenames should only contain the filename and
     /// extension, e.g. "unacknowledged-writes.json". Filenames are matched case-sensitively.
-    pub(crate) fn skip_files(self, skipped_files: &'static [&'static str]) -> Self {
+    pub(crate) fn skip_files(self, skipped_files: impl Into<Option<Vec<&'static str>>>) -> Self {
```

Comment:
I think this could be slightly improved (and avoid the noise of `vec!` where it's not needed) by having the signature be
`fn skip_files(self, skipped_files: &[&'static str]) -> Self`
and then storing it via
`skipped_files: Some(skipped_files.to_vec())`.

Reply:
The benefit of using vecs comes at the time of construction: when we need to skip conditionally based on a feature or server version/topology, a vec lets us create a list and then append to it rather than create two distinct lists that might have duplication (e.g. see the diff for `valid_pass` in this file).

Reply:
Right - the signature I propose doesn't have the `'static` bound that the original `skip_files` had on the slice, so those cases can just pass `&local_vec` (or `local_vec.as_slice()` if type inference is confused).

Reply:
Ah I see what you mean - good idea. updated